### PR TITLE
Feat/4/register a right

### LINKED ISSUE #4: Register a Right
Parent: https://github.com/bigchaindb/org/issues/17

### PR DESCRIPTION
**Pull request by TimDaub**
@sohkai Up to you what you do with this.

Fixes #4.

### CODE REVIEW

**Comment by sohkai** [comment]
Will require #8 to pass all tests.

**Comment by sohkai** [comment]
@TimDaub now reviewable

File: README.md [hunk 1]
```diff
@@ -241,3 +241,49 @@ or
...
+Rather than registering new Rights directly to resulting rightsholders, new
+Rights are initially registered to the current holder of the existing Right or
+Copyright that allows for the new Right's creation. Doing so allows the initial
+link between the source rightsholder and resulting rightsholder to be kept as
+part of the new Right's chain of provenance.
+
+Note that the attributes for the `right` may be much more diverse; see the COALA
+IP models definition (not yet publicly available yet).
+
+Also see [transfering a Right](FIXME) on how to transfer a registered Right to
```
Comment (TimDaub) [inline README.md]: Maybe don't link yet

File: README.md [hunk 2]
```diff
@@ -241,3 +241,49 @@ or
...
+new holders.
+
+```
+POST /api/v1/right/
```
Comment (TimDaub) [inline README.md]: URLs to endpoints should always be in plural form: `rights`.

I just noticed that we've made that mistake with manifestation as well :/

Comment (sohkai) [reply README.md]: Oops...

File: web/utils.py [hunk 3]
```diff
@@ -32,3 +34,16 @@ def get_bigchaindb_api_url():
...
+
+def parse_model(required_fields):
+    def _parse_model(inputs):
+        for field in required_fields:
+            try:
+                value = inputs[field]
+            except KeyError:
+                raise KeyError('`{}` must be provided'.format(field))
+            if bool(value) is not True:
+                raise ValueError("`{}`'s value must be defined")
```
Comment (TimDaub) [inline web/utils.py]: I think the `.format` went missing here?

Comment (TimDaub) [reply web/utils.py]: It actually was never there :thinking:

Comment (sohkai) [reply web/utils.py]: Turns out it was never there. Fixed.

File: web/views/right.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+        right = coalaip.derive_right(right_data=right,
+                                     current_holder=current_holder)
+
+        res = {'right': right.to_jsonld()}
+
+        return res
+
+
+right_api.add_resource(RightApi, '/rights', strict_slashes=False)
```
Comment (TimDaub) [inline web/views/right.py]: In the docs you've listed `/right` as the URI.

My recommendation: Keep it as `/rights`, change the docs to the correct URI and then either me or you do a follow up for the manifestation endpoint.

Comment (sohkai) [reply web/views/right.py]: Had a thought about renaming the file to be plural too, since right now it's just `right.py` and `manifestation.py`. Thoughts?

**Comment by TimDaub** [comment]
LGTM

**Pull request merged**